server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3547,6 +3547,8 @@ private Ternary<List<Long>, Integer, String[]> searchForDiskOfferingsIdsAndCount
 
         if (state != null) {
             sc.setParameters("state", state);
+        } else {
+            sc.setParameters("state", Arrays.asList(ServiceOffering.State.Active, ServiceOffering.State.Inactive));
```

Comment:
Why is this required?

Reply:
in case we don't specify what offering is required , this would list offerings in both active and inactive states

Reply:
If you do not set the state condition (AKA `where state in x`), both active and inactive will be listed.